lua/mediaplayer/services/youtube/init.lua

```diff
@@ -134,18 +135,96 @@ function SERVICE:GetMetadata( callback )
 		callback(self._metadata)
 
 	else
-
 		local videoId = self:GetYouTubeVideoId()
-		local apiurl = MetadataUrl:format( videoId, APIKey )
+		local videoUrl = "https://www.youtube.com/watch?v="..videoId
 
-		self:Fetch( apiurl,
+		self:Fetch( videoUrl,
+			--On Success
 			function( body, length, headers, code )
-				OnReceiveMetadata( self, callback, body )
+				local metadata = self:ParseYTMetaDataFromHTML(body, videoId)
+
+				--html couldn't be parsed
+				if (!metadata.title || !metadata.duration) then
+					callback(false, "Failed to parse HTML Page for metadata")
+					return
+				end
+
+				self:SetMetadata(metadata, true)
+
+				if self:IsTimed() then
+					MediaPlayer.Metadata:Save(self)
+				end
+			
+				callback(self._metadata)
 			end,
+			-- On failure
 			function( code )
 				callback(false, "Failed to load YouTube ["..tostring(code).."]")
-			end
+			end,
+			--Headers
+			{
+				["User-Agent"] = "Googlebot"
+			}
 		)
-
 	end
 end
+
+---
+-- Function to parse video metadata straight from the html instead of using the API
+--
+function SERVICE:ParseYTMetaDataFromHTML( html, videoId )
+
+    -- Get the value for an attribute from a html element
+    local function ParseElementAttribute( element, attribute )
+        if !element then return end
+        -- Find the desired attribute
+        local output = string.match( element, attribute.."%s-=%s-%b\"\"" )
+        if !output then return end
+        -- Remove the 'attribute=' part
+        output = string.gsub( output, attribute.."%s-=%s-", "" )
+        -- Trim the quotes around the value string
+        return string.sub( output, 2, -2 )
+    end
+    -- Get the contents of a html element by removing tags
+    -- Used as fallback for when title cannot be found
+    local function ParseElementContent( element )
+        if !element then return end
+        -- Trim start
+        local output = string.gsub( element, "^%s-<%w->%s-", "" )
+        -- Trim end
+        return string.gsub( output, "%s-</%w->%s-$", "" )
+	end
+	
+	--MetaData table to return when we're done
+	local metadata = {}
+    
+    -- Lua search patterns to find metadata from the html
+    local patterns = {
+        ["title"] = "<meta%sproperty=\"og:title\"%s-content=%b\"\">",
+        ["title_fallback"] = "<title>.-</title>",
+        ["thumb"] = "<meta%sproperty=\"og:image\"%s-content=%b\"\">",
+        ["thumb_fallback"] = "<link%sitemprop=\"thumbnailUrl\"%s-href=%b\"\">",
+        ["duration"] = "<meta%sitemprop%s-=%s-\"duration\"%s-content%s-=%s-%b\"\">",
+        ["live"] = "<meta%sitemprop%s-=%s-\"isLiveBroadcast\"%s-content%s-=%s-%b\"\">"
+    }
```

Comment:
This should be moved outside of and above the `SERVICE:ParseYTMetaDataFromHTML` function definition.

Reply:
Same with the `Parse*` functions